src/components/application_manager/src/commands/mobile/get_way_points_request.cc

```diff
@@ -32,18 +39,102 @@ void GetWayPointsRequest::Run() {
                  true);
 }
 
+bool GetWayPointsRequest::CheckResponseSyntax(
+    const smart_objects::SmartObject& obj) {
+  const char* str = obj.asCharArray();
+  const bool is_str_empty = str && !str[0];
+  return is_str_empty || CheckSyntax(str);
+}
+
+bool GetWayPointsRequest::ValidateSmartMapResponse(
+    const smart_objects::SmartObject& obj, ConstSetIter key) {
+  const smart_objects::SmartObject& value = obj[*key];
+  switch (value.getType()) {
+    case SmartType_Array:
+    case SmartType_Map: {
+      return ValidateResponseFromHMI(value);
+    }
+    case SmartType_String: {
+      return CheckResponseSyntax(value);
+    }
+    default:
+      // Catch SmartType_Integer or SmartType_Double
+      break;
+  }
+  return true;
+}
+
+bool GetWayPointsRequest::ValidateResponseFromHMI(
+    const smart_objects::SmartObject& obj) {
+  switch (obj.getType()) {
+    case SmartType_Array: {
+      for (uint32_t i = 0; i < obj.length(); ++i) {
+        if (!ValidateResponseFromHMI(obj[i])) {
+          return false;
+        }
+      }
+      break;
+    }
+    case SmartType_Map: {
+      const std::set<std::string> keys = obj.enumerate();
+      for (ConstSetIter key = keys.begin(); key != keys.end(); key++) {
+        if (!ValidateSmartMapResponse(obj, key)) {
+          return false;
+        }
+      }
+      break;
+    }
+    case SmartType_String: {
+      return CheckResponseSyntax(obj);
+    }
+    default:
+      // Catch SmartType_Integer or SmartType_Double
+      break;
+  }
+  return true;
+}
+
+void GetWayPointsRequest::onTimeOut() {
+  SDL_AUTO_TRACE();
```

Comment:
@OHerasym, I think this method will never called, because all invalid response will be  handled by `application_manager_impl.cc` method   `HandleWrongMessageType` . Please try reproduce defect without `GetWayPointsRequest::onTimeOut`.